pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -81,12 +85,19 @@ func (a *genericActuator) Reconcile(ctx context.Context, worker *extensionsv1alp
 		// When the Shoot gets hibernated we want to remove the cluster auto scaler so that it does not interfer
 		// with Gardeners modifications on the machine deployment's replicas fields.
 		if controller.IsHibernated(cluster.Shoot) || rollingUpdate {
-			deployment := &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Namespace: worker.Namespace, Name: gardencorev1alpha1.DeploymentNameClusterAutoscaler}}
-			if err := a.client.Delete(ctx, deployment); err != nil && !apierrors.IsNotFound(err) {
-				return err
+			deployment := &appsv1.Deployment{}
+			if err := a.client.Get(ctx, kutil.Key(worker.Namespace, gardencorev1alpha1.DeploymentNameClusterAutoscaler), deployment); err != nil {
+				if !apierrors.IsNotFound(err) {
+					return err
+				}
+			} else {
+				if err := util.ScaleDeployment(ctx, a.client, deployment, 0); err != nil {
+					return err
+				}
 			}
-			if err := util.WaitUntilResourceDeleted(ctx, a.client, deployment, 5*time.Second); err != nil {
-				return err
+
+			if rollingUpdate {
```

Comment:
```suggestion
mustScaleUpClusterAutoscaler = rollingUpdate
```

I would also move to either line 84 or 156. Or leave out `mustScaleUpClusterAutoscaler` and use `rollingUpdate` instead.